### PR TITLE
fix(buttons): outline variations missing style with wp 6.6

### LINKED ISSUE #217: [WP 6.6] Outline buttons seem broken
Maybe it'll be fixed before 6.6 is officially launched but the outline buttons look broken using 6.6-RC1.

__6.5__

__6.6-RC1__

### PR DESCRIPTION
**Pull request by thomasguillot**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-block-theme/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

With WP 6.6, the outline button styles disappeared in the editor. For some reason, the styles applied via theme.json aren't being displayed in the back-end. This PR adds the background/border/color default properties to the scss file.

There's also a small fix for the donate block so the outline buttons on `:hover` matches  the style of this theme.

Closes #217 .

### How to test the changes in this Pull Request:

1. Add a buttons block, with different buttons in it, including outline.
2. Notice in the editor the outline button is black by default (it should be the accent color)
3. But in the front-end it's fine
4. Add a donate block, use a tiered layout with the modern style.
5. Switch to this branch and check the front-end and back-end.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: src/scss/blocks/_button.scss [hunk 1]
```diff
@@ -57,6 +58,10 @@
 	&.is-style-outline {
 		> .wp-block-button__link,
 		form .wp-block-button__link {
+			background: var(--wp--preset--color--base);
+			border: 1px solid currentcolor;
```
Comment (chickenn00dle) [inline src/scss/blocks/_button.scss:62]: Question: Do we need to make changes to the border color on hover as well? Solid buttons seem to become a darker shade of the same color on hover, while outline buttons appear to become black.

**Review by chickenn00dle — COMMENTED** [review]
Hey @thomasguillot,

The changes for the button block look good on my end.

I did run into an issue with the donate block submit button, although this issue seems to exist in other themes. In any case, I'll go ahead and mention it here for posterity. For some reason the donate block submit button background color can't be cleared on my end. I can make the change in the editor, but after saving and refreshing, the change is reverted, and the front end always shows the initial background color selection:

Editor:
Screenshot 2024-07-17 at 15 13 57

Front end:
Screenshot 2024-07-17 at 15 14 23

Aside from the above which is not a blocker, I did have one question before giving this a thumbs up.

**Comment by thomasguillot** [comment]
> For some reason the donate block submit button background color can't be cleared on my end. I can make the change in the editor, but after saving and refreshing, the change is reverted, and the front end always shows the initial background color selection

Yeah I think it's because the red is hard-coded. Frankly, I don't like this behaviour and we should probably change it because what you're experienced is a bug to me.

I also think the colours should be reworked so we can use variables. At the moment, even if I select my accent colour, it uses the HEX value instead of the variable which means we can't use variables for a pattern.

E.g.
https://github.com/search?q=repo%3AAutomattic/newspack-blocks%20%236c6c6c&type=code where I wanted to use `var(--wp--preset--color--contrast-3)` instead

File: src/scss/blocks/_button.scss (hunk 1 above)
Comment (thomasguillot) [reply src/scss/blocks/_button.scss:62]: With the Block Theme, the border-color of the outline-button should match the text-color. And on `:hover`, the text should become `var(--wp--preset--color--contrast)` and the background `var(--wp--preset--color--base-2)`

Comment (chickenn00dle) [reply src/scss/blocks/_button.scss:62]: Thanks for clarifying!

**Review by chickenn00dle — APPROVED** [review]
> Yeah I think it's because the red is hard-coded. Frankly, I don't like this behaviour and we should probably change it because what you're experienced is a bug to me.

I feel the same. I went ahead and created a new issue for it here:
1200550061930446-as-1207849443195902/f

**Pull request merged**